app/services/vimeo/api_service.rb

```diff
@@ -29,21 +33,13 @@ def create_video_resource(size)
     end
 
     def add_whitelist_urls(video_id)
-      data = {
-        privacy: {
-          embed: 'whitelist'
-        }
-      }
-
-      @current_school.domains.pluck(:fqdn).map do |fqdn|
-        put("/videos/#{video_id}/privacy/domains/#{fqdn}", data)
-      end
+      Vimeo::SetEmbedLinks.perform_later(@current_school, video_id)
```

Comment:
Since this method is just a single line, it would be cleaner to move the `Vimeo:: AddAllowedDomainsToVideo.perform_later(...)` line into the `create_video` method.

I think an even better option would be to do that _outside_, in the mutator class, so that _that_ call-site makes it clear which API operations are happening, and in what order (one synchronous API call, and one asynchronous job queuing).

```ruby
class CreateVimeoVideoMutator < ApplicationQuery
  # ...
  def create_vimeo_video
    vimeo_api = Vimeo::ApiService.new(current_school)
    response = vimeo_api.create_video(size)
    video_id = response[:uri].split('/')[-1]
    Vimeo:: AddAllowedDomainsToVideo.perform_later(current_school, video_id)

    { 
      upload_link: response[:upload][:upload_link],
      uri: response[:uri]
    }
  end
```

That way, the `ApiService#create_video` method call does not have any _side-effects_ - it remains simple, only doing one thing, and the mutator's `create_vimeo_video` method clearly does two things - creating the video, and then adding allowed domains.